src/ua/main.cpp

```diff
@@ -184,6 +187,17 @@ void compressChunks() {
   }
 }
 
+bool is_chunk_complete(Chunk *c) {
+    bool chunk_complete = false;
+    JSON result = fileDescribe(c->fileID);
+    string partIndex = boost::lexical_cast<string>(c->index + 1); // minimum part index is 1
+
+    if (result["parts"].has(partIndex) && result["parts"][partIndex]["state"].get<string>() == "complete")
```

Comment:
I think this could be more simply expressed:

`return result["parts"].has(partIndex) && result["parts"][partIndex]["state"].get<string>() == "complete"`